inst/scribble/scribble.js

```diff
@@ -0,0 +1,626 @@
+    const b = parseInt(color.substring(4, 6), 16) // hexToB
+    const score = ((r * 0.299) + (g * 0.587) + (b * 0.114))
+    return (score > threshold) ? dark : light
+  }
+
+  assembleToolbox () {
+    this.createToolbox()
+
+    // Build draw, erase, clear buttons
+    this.drawBtn = this.createButton('draw', 'draw', 'Start Drawing (S)')
+    this.eraseBtn = this.createButton('erase', 'eraser', 'Erase Lines')
+    this.clearBtn = this.createButton('clear', 'trash', 'Clear This Slide')
+
+    this.colorPicker = this.createColorPicker()
+    this.colorPicker.addEventListener('input', () => {
+      this.currColor = this.colorPicker.value
+      this.currFabric.freeDrawingBrush.color = this.currColor
+      this.setPenColorCSSVariables(this.currColor)
+    })
+
+    ;[this.drawBtn, this.eraseBtn, this.clearBtn, this.colorPicker]
+      .forEach((btn) => {
+        this.toolBox.appendChild(btn)
+      })
+
+    ;['click', 'touchend'].forEach((action) => {
+      this.toolBox.addEventListener(action, ev => ev.stopPropagation())
+    })
+  }
+
+  addToolboxToSlide () {
+    const canvasDiv = this.getVisibleSlideOuterContainer()
+    canvasDiv.appendChild(this.toolBox)
+    if (this.hideToolbox) {
+      this.toolBox.classList.add('minimized')
+    }
+  }
+
+  addToggleToolbox () {
+    const self = this
+
+    document.addEventListener('keydown', (ev) => {
+      if (ev.keyCode === this.launchKey) {
+        self.toggleToolbox()
+      }
+    })
+
+    this.toolBox.addEventListener('mouseleave', function (ev) {
+      if (self.drawMode | self.eraseMode) {
+        return
+      }
+      self.minimizeTimeout = setTimeout(() => self.toggleToolbox(false), 2000)
+    })
+
+    this.toolBox.addEventListener('mouseenter', function (ev) {
+      if (self.minimizeTimeout) {
+        clearTimeout(self.minimizeTimeout)
+      }
+    })
+  }
+
+  toggleToolbox (show) {
+    const isMinimized = this.toolBox.matches('.minimized')
+    if (show && show === !isMinimized) return
+
+    this.hideToolbox = !(show || isMinimized)
+    this.toolBox.classList.toggle('minimized')
+
+    if (this.hideToolbox) {
+      this.toolBox.classList.add('minimized')
+      if (this.drawMode) {
+        this.drawBtn.click()
+      } else if (this.eraseMode) {
+        this.eraseBtn.click()
+      }
+    } else {
+      this.toolBox.classList.remove('minimized')
+      clearTimeout(this.minimizeTimeout)
+      this.minimizeTimeout = null
+      if (!this.drawMode) this.drawBtn.click()
+    }
+  }
+
+  addCanvasHelpText () {
+    const helpTable = document.querySelector(
+      '.remark-help-content table.light-keys'
+    )
+    if (!helpTable) {
+      console.error(
+        'Could not find remark help table, has remark been initialized?'
+      )
+      return
+    }
+    const newRow = document.createElement('tr')
+    newRow.innerHTML += '<td><span class="key">s</span></td>'
+    newRow.innerHTML += '<td>Toggle scribble toolbox</td>'
```

Comment:
Pedantic: We could change this to `Toggle scribble tools` if we've gone away from calling it a toolbox. Super minor.

Reply:
I like toolbox and am cool with keeping the name. It's still "toolbox" in the docs, did the name change somewhere else?